### PR TITLE
modified pages for female prison ids

### PR DESCRIPTION
**Pull request by Olusola-Opencast**

### CODE REVIEW

File: server/views/pages/tasklist.html [hunk 1]
```diff
@@ -70,6 +70,7 @@ <h1 class="govuk-heading-m">Prisoner details</h1>
                 <div class="govuk-heading-m">Offending history</div>
                 <div class="govuk-!-margin-top-2">{% if offendingHistoryCompleted %}Completed{% else %}Not yet checked{% endif %}</div>
             </li>
+            {% if not data.isFemale %}
             <li class="app-task-list__item">
```
Comment (kberec) [inline server/views/pages/tasklist.html:74]: As this is now disabled, going forward db values won't be populated anymore.
Have you searched for every occurrence of the data saved off here, to make sure it is nullsafe if anywhere else is used?

File: test/routes/tasklist.test.js [hunk 2]
```diff
@@ -85,10 +89,34 @@ describe('GET /tasklist/', () => {
         expect(res.text).toMatch(/Digital Prison Services.+Categorisation dashboard/s)
         expect(res.text).toContain('Categorisation task list')
         expect(res.text).toContain('Offending history')
+        expect(res.text).toContain('Further charges')
         expect(res.text).toContain('Not yet checked')
         expect(res.text).toContain('Conditional Release Date')
         expect(res.text).toContain('04/04/2020')
-      }))
+        expect(data.isFemale).toContain(false)
```
Comment (kberec) [inline test/routes/tasklist.test.js:96]: This is checking the data you set up. Same below with the other test.
Also constant data is unrelated, i.e. if you change value to true, test will still pass. Happy to discuss.

**Review by kberec — DISMISSED** [review]

File: server/routes/tasklist.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const asyncMiddleware = require('../middleware/asyncMiddleware')
 const Status = require('../utils/statusEnum')
 const CatType = require('../utils/catTypeEnum')
 const { addSocProfile, inProgress } = require('../utils/functionalHelpers')
-const { get10BusinessDays } = require('../utils/utils')
+const { get10BusinessDays, isFemalePrisonId } = require('../utils/utils')
```
[the remark this answers is not in the document]
Comment (Olusola-Opencast) [reply server/routes/tasklist.js:7]: This doesn't work as it breaks the jest tests.

[the remark this answers is not in the document]
Comment (Olusola-Opencast) [reply server/routes/tasklist.js]: sorted

File: server/views/pages/tasklist.html (hunk 1 above)
Comment (Olusola-Opencast) [reply server/views/pages/tasklist.html:74]: The database string isn't there if further charges isn't filled in and isn't called anywhere.

File: test/routes/tasklist.test.js (hunk 2 above)
Comment (Olusola-Opencast) [reply test/routes/tasklist.test.js]: done - painful.

File: test/routes/tasklist.test.js [hunk 4]
```diff
@@ -76,24 +77,76 @@ afterEach(() => {
 })
 
 describe('GET /tasklist/', () => {
-  test('should render tasklist', () =>
-    request(app)
+  test('should render a tasklist for male prison', () => {
+    userService.getUser.mockResolvedValue({
```
Comment (kberec) [inline test/routes/tasklist.test.js:81]: It still works without this mock where 'female: false'. Same down in lines 133, 205, 228. Is this mock unnecessary?

File: test/routes/tasklist.test.js [hunk 5]
```diff
@@ -162,12 +205,13 @@ describe('GET /tasklist/', () => {
       })
   })
 
-  test('should render categoriserSubmitted page', () =>
+  test('should render categoriserSubmitted page', () => {
     request(app)
       .get('/categoriserSubmitted/12345')
-      .expect(200)
+      .expect(500)
```
Comment (kberec) [inline test/routes/tasklist.test.js:211]: .expect(200) should work

**Review by kberec — APPROVED** [review]

**Pull request merged**